### ISSUE TITLE
Core libdefs can't really specify platform-defined getters/setters

### ISSUE THREAD

**Issue by jeffmo** [facebook/flow#3426]
As described over in #3328, putting getter/setter fields in anything in a core libdef is problematic since Flow doesn't support getters/setters well (or at all, out of the box).

We have a few ways we could address this:

1. Support getters/setters better. This is really hard since getters/setters are pretty dangerous for sound checking without effects tracking
2. Map getters/setters back to normal fields in a libdef if the `enable_unsafe_getters_and_setters` is set to `false`. This would let us faithfully describe the real world of platform APIs, but draw a line for graceful handling of more robust user code

**Comment by samwgoldman** [comment]
When Gabe initially added support for getters and setters, we used to treat them identically to fields. Now we actually distinguish between fields and get/set methods, which means we can actually havoc refinements when they are read from and written to (which was the unsound part, iirc).

Another motivation for supporting get/set out-of-the-box: spread support will require that we know the object layout for instance types. Since class properties are own, any getter that is modeled as a class prop will be considered own and spread in. (We also need to consider enum, which is related, but don't want to drag this issue down with that.)

**Comment by jeffmo** [comment]
> we can actually havoc refinements

In theory yes, but it would be pretty painful to have every property access havoc :/

**Comment by samwgoldman** [comment]
No, but Avik has a now-pretty-old diff that I think would enable it..
